apps/api/src/domain/models/audio-item/commands/translate-audio-item-name/translate-audio-item-name.command-handler.ts

```diff
@@ -0,0 +1,48 @@
+import { CommandHandler, ICommand } from '@coscrad/commands';
+import {
+    MultilingualTextItem,
+    MultilingualTextItemRole,
+} from '../../../../../domain/common/entities/multilingual-text';
+import { Valid } from '../../../../../domain/domainModelValidators/Valid';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { InMemorySnapshot } from '../../../../../domain/types/ResourceType';
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { ResultOrError } from '../../../../../types/ResultOrError';
+import { BaseUpdateCommandHandler } from '../../../shared/command-handlers/base-update-command-handler';
+import { BaseEvent } from '../../../shared/events/base-event.entity';
+import { AudioItem } from '../../entities/audio-item.entity';
+import { AudioItemNameTranslated } from './audio-item-name-translated-event';
+import { TranslateAudioItemName } from './translate-audio-item-name.command';
+
+@CommandHandler(TranslateAudioItemName)
+export class TranslateAudioItemNameCommandHandler extends BaseUpdateCommandHandler<AudioItem> {
+    protected async fetchRequiredExternalState(
+        _: TranslateAudioItemName
+    ): Promise<InMemorySnapshot> {
+        return Promise.resolve(new DeluxeInMemoryStore({}).fetchFullSnapshotInLegacyFormat());
+    }
+
+    protected validateExternalState(
+        _state: InMemorySnapshot,
+        _instance: AudioItem
+    ): InternalError | Valid {
+        return Valid;
+    }
+
+    protected actOnInstance(
+        AudioItem: AudioItem,
+        { text, languageCode }: TranslateAudioItemName
+    ): ResultOrError<AudioItem> {
+        return AudioItem.translateName(
+            new MultilingualTextItem({
```

Comment:
You should be able to leave off the `new MultilingualTextItem()` part and just pass the DTO 
```ts
        return AudioItem.translateName(
            {
                text,
                languageCode,
                role: MultilingualTextItemRole.freeTranslation,
            })
```

Reply:
done